Source/BoundaryConditions/ERF_FillPatcher.cpp

```diff
@@ -62,87 +57,106 @@ void ERFFillPatcher::Define (BoxArray const& fba, DistributionMapping  fdm,
-            bl_subset.join(bndry);
-        }
-    }
-
-    // Coarse box list (coincides with grown fine box list)
+    // Coarse box list
     BoxList cbl;
     cbl.set(m_ixt);
-    cbl.reserve(bl.size());
-    for (auto const& b : bl) {
-        cbl.push_back(interp->CoarseBox(b, m_ratio));
+    cbl.reserve(fba.size());
+    for (int i(0); i<fba.size(); ++i) {
+        cbl.push_back(interp->CoarseBox(fba[i], m_ratio));
     }
 
-    // Box arrays for fine and coarse
-    BoxArray cf_fba(std::move(bl));
+    // Box arrays for the coarse data
     BoxArray cf_cba(std::move(cbl));
-    BoxArray cf_fba_s(std::move(bl_subset));
-    DistributionMapping cf_dm(cf_fba);
-    DistributionMapping cf_dm_s(cf_fba_s);
+    DistributionMapping cf_dm(cf_cba);
 
+    // Two coarse patches to hold the data to be interpolated
+    m_cf_crse_data_old = std::make_unique<MultiFab> (cf_cba, fdm, m_ncomp, 0);
+    m_cf_crse_data_new = std::make_unique<MultiFab> (cf_cba, fdm, m_ncomp, 0);
 
-    // Fine patch to hold the time-interpolated state
-    m_cf_fine_data = new MultiFab (cf_fba, cf_dm, m_ncomp, 0);
-
-    // Fine subset patch to hold the time-interpolated state
-    m_cf_fine_subset_data = new MultiFab (cf_fba_s, cf_dm, m_ncomp, 0);
+    // Integer masking array
+    m_cf_mask = std::make_unique<iMultiFab> (fba, fdm, 1, 0);
+    m_cf_mask->setVal(m_relax_mask);
 
-    // Two coarse patches to hold the data to be interpolated
-    m_cf_crse_data[0] = new MultiFab (cf_cba, cf_dm, m_ncomp, 0);
-    m_cf_crse_data[1] = new MultiFab (cf_cba, cf_dm, m_ncomp, 0);
+    // Populate mask array
+    if (nghost_set < 0) {
+        m_cf_mask->setVal(m_set_mask);
+        BuildMask(fba,nghost_set,m_set_mask-1);
+    }
+    BuildMask(fba,nghost,m_relax_mask-1);
 }
 
+void ERFFillPatcher::BuildMask (BoxArray const& fba,
+                                int nghost,
+                                int mask_val)
+{
+    // Minimal bounding box of fine BA plus a halo cell
+    Box fba_bnd = amrex::grow(fba.minimalBox(), IntVect(1,1,0));
+
+    // BoxList and BoxArray to store complement
+    BoxList com_bl; BoxArray com_ba;
+
+    // Compute the complement
+    fba.complementIn(com_bl,fba_bnd);
+
+    // com_bl cannot be null since we grew with halo cells
+    AMREX_ALWAYS_ASSERT(com_bl.size() > 0);
+
+    // Grow the complement boxes and trim with the bounding box
+    Vector<Box>& com_bl_v = com_bl.data();
+    for (int i(0); i<com_bl.size(); ++i) {
+        Box& bx = com_bl_v[i];
+        bx.grow(IntVect(-nghost,-nghost,0));
+        for (int idim(0); idim<AMREX_SPACEDIM; ++idim) {
+            if (bx.bigEnd(idim) > fba_bnd.bigEnd(idim)) bx.setBig(idim,fba_bnd.bigEnd(idim));
+            if (bx.smallEnd(idim) < fba_bnd.smallEnd(idim)) bx.setSmall(idim,fba_bnd.smallEnd(idim));
+        }
+    }
+
+    // Do second complement with the grown boxes
+    com_ba.define(com_bl);
+    com_ba.complementIn(com_bl, fba_bnd);
+
+    // Fill mask based upon the com_bl BoxList
```

Comment:
There is a more efficient way to do this. Since this pattern of setVal in a BoxArray's region can be useful in other places, I will add a function to amrex. After that is merged, we can do a new PR to use it.